File: app/src/main/java/com/klinker/android/twitter_l/data/roomdb/daos/ActivityDao.java
package com.klinker.android.twitter_l.data.roomdb.daos;


import com.klinker.android.twitter_l.data.roomdb.entities.Activity;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

@Dao
public abstract class ActivityDao {

    @Insert
    abstract void insertActivitySpecificInfo(Activity activity);

    @Query("SELECT * FROM activities WHERE account = :account")
    abstract Activity getAllActivities(int account);

}

File: app/src/main/java/com/klinker/android/twitter_l/data/roomdb/daos/ActivityDao.kt
package com.klinker.android.twitter_l.data.roomdb.daos


import com.klinker.android.twitter_l.data.roomdb.entities.Activity

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
abstract class ActivityDao {





    //abstract fun getDisplayActivities(account: Int, page : Int = 1, pageSize: Int = 60);

    @Query("SELECT id FROM activities WHERE account = :account ORDER BY time DESC LIMIT :size")
    abstract fun getLatestIds(account: Int, size : Int = 2) : List<Long>

    @Query("DELETE FROM activities WHERE account = :account")
    abstract fun deleteAllActivities(account: Int)

    @Query("DELETE FROM activities WHERE id NOT IN(SELECT id FROM activities WHERE account = :account ORDER BY time DESC LIMIT :trimSize)")
    abstract fun trimDatabase(account: Int, trimSize: Int)

}
